src/stream-tcp-private.h

```diff
@@ -232,13 +232,12 @@ enum TcpState
                                             (p)->flowflags |= FLOW_PKT_TOCLIENT : (p)->flowflags &= ~FLOW_PKT_TOCLIENT \
                                             (p)->flowflags |= FLOW_PKT_TOSERVER : 0)
 
-/* Macro's for comparing Sequence numbers
- * Page 810 from TCP/IP Illustrated, Volume 2. */
```

Comment:
The goal is to compare TCP sequence numbers, there `1` is bigger than `4200000000`. These macros do that. Please see the reference above.

Reply:
In StreamTcpValidateAck, will compare with SEQ_GT(ack, stream->last_ack) && SEQ_LEQ(ack, stream->next_win), my test case:
ack is 986354499, stream->next_win is 3199576431, SEQ_LEQ(ack, stream->next_win) should be True, but by the original macros is False, because unsigned ack -unsigned next_win is unsigned result, will large than zero.

Reply:
I think this is correct wrt the macro. It considers `ack` to be ahead of `next_win`, so GT. The macros are designed this way to deal with the wrapping around the sequence numbers do.

You may have found another issue, but I'm pretty sure the macros work as expected.

Reply:
Mybe you can run this pcap to test : https://github.com/mtimebombm/suricata/blob/master/imap-server.pcap, 
The above problem will occur in the third package.

Maybe something went wrong elsewhere， but i don't think unsigned parameter subtraction unsigned work as expected.I find the macro from BSD source in tcp_seque.h,  but they used this macros by singed parameters.

Reply:
Thanks for the pcap, I've identified the issue and one other one as well. Will push a PR soon.

Ticket: https://redmine.openinfosecfoundation.org/issues/4228